codalab/worker/singularity_utils.py

```diff
@@ -0,0 +1,16 @@
+def is_shub(img):
+    return img.startswith("shub://")
+
+
+def is_sylabs(img):
+    return img.startswith("library://")
+
+
+def is_singularity(image_spec):
+    # sylabs container registry or (deprecated) singlarity hub
+    return is_shub(image_spec) or is_sylabs(image_spec)
+
+
+def get_singularity_container_size(image_spec):
```

Comment:
This is used, but not implemented

Reply:
If ever shub or sylabs will start supporting querying image sizes (which they said is one of their goals), this function will be useful. I think we should keep it here, as it will be also useful to unify the logic in the base class.

Reply:
Not sure if there's value in just keeping it here and leaving it unimplemented. If you really want to keep it, maybe raise a NotImplementedError and add a docstring explaining why?